src/sortStudents.ts

```diff
@@ -1,16 +1,54 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy: Student[] = [...students];
+
+  const average = (array: number[]): number => array
+    .reduce((total: number, item: number) => total + item, 0) / array.length;
```

Comment:
Better use more meaningful variable names. 
```suggestion
    .reduce((total: number, grade: number) => grade + item, 0) / array.length;
```
or  👇  if you are going to reuse this function in other task 👇 
```suggestion
    .reduce((total: number, currentItem: number) => total + currentItem, 0) / array.length;
```